components/src/dynamo/planner/utils/planner_config.py

```
@@ -0,0 +1,206 @@
+            try:
+                data = json.loads(text)
+            except json.JSONDecodeError:
+                try:
+                    import yaml
+
+                    data = yaml.safe_load(text)
+                except ImportError:
+                    raise ValueError(
+                        f"Could not parse config file '{path}'. "
+                        "For YAML support, install pyyaml."
+                    )
```

Ruff (0.15.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)